src/latexify/codegen/expression_rules_test.py

```diff
@@ -0,0 +1,25 @@
+"""Tests for latexify.codegen.expression_rules."""
+
+from __future__ import annotations
+
+import ast
+
+import pytest
+
+from latexify.codegen import expression_rules
+
+
+@pytest.mark.parametrize(
+    "tree,precedence",
+    [
+        (ast.Call(), expression_rules._CALL_PRECEDENCE),
+        (ast.BinOp(op=ast.Add()), expression_rules._PRECEDENCES[ast.Add]),
+        (ast.UnaryOp(op=ast.UAdd()), expression_rules._PRECEDENCES[ast.UAdd]),
+        (ast.BoolOp(op=ast.And()), expression_rules._PRECEDENCES[ast.And]),
+        (ast.Compare(ops=[ast.Eq()]), expression_rules._PRECEDENCES[ast.Eq]),
+        (ast.Name(), expression_rules._INF_PRECEDENCE),
+        (ast.Attribute(), expression_rules._INF_PRECEDENCE),
+    ],
+)
+def test_get_precedence(tree: ast.AST, precedence: int) -> None:
```

Comment:
Half the tests call this argument `tree` and the other half `node`. Personally I prefer `node` but don't really care but would make it easier to understand tests if the same variable name was used across them.

Reply:
renamed to `node` anyway since the name have been used in codegens. `tree` is used in frontend though. I prefer this name a bit because it slightly means "entire tree" rather than the fragment. (But this is just a bike-shedding problem I guess)